### ISSUE TITLE
Allow SFR cellid to be (0, 0, 0) or `NONE` for easier parsing

### ISSUE THREAD

**Issue by ghost** [MODFLOW-USGS/modflow6#447]
Using `NONE` to identify unconnected SFR cells makes it impossible to read in a packagedata file assuming columns (e.g. using Pandas or excel) for a structured grid because it changes the number of columns by 2 (usually is i,j,k identifying cells but becomes a single column when NONE is the identifier). This could be fixed by allowing a negative index (for example -99) for all three indices like -99 -99 -99 as an option.

**Comment by emorway-usgs** [comment]
@mnfienen-usgs perhaps not ideal, but one work around for reaches not connected to the aquifer might be to assign dummy cell indices (e.g., 1,1,1) and set the hydraulic conductivity of the streambed equal to 0.0.  I think that should shut-off the stream-aquifer exchange as well.

**Comment by aleaf** [comment]
I like the idea of having a null connection integer value in MODFLOW-6 instead of NONE. In addition to issues with reading tables from files, the jagged structure also means that nice pythonic stuff like unpacking
```python
cellid = [(1, 2, 3), (1, 2, 3), (1, 2, 3),'none']
k, i, j = zip(*cellid)
```
doesn't work. You could do the above if ``(None, None, None)`` were allowed, but then the resulting k, i, j sequences can't be used for indexing or cast to int arrays without first replacing the `NoneType` (or `'none'`) values. A null connection integer could be used in a tuple for structured grids, or as-is in unstructured grids.

**Comment by langevin-usgs** [comment]
Is this a generic capability needed throughout MF6?  Or is this just needed for SFR because of the NONE connection?  We've tried to make many of the input blocks have a regular structure that can be read by Pandas, for example, but it is hasn't been part of the fundamental design criteria because some datasets are inherently jagged.

**Comment by mnfienen-usgs** [comment]
I understand some data files are inherently jagged, so at this point I'm focused on SFR specifically. Maybe @emorway-usgs suggestion could work. Just seems that a minor tweak in the format for SFR would make a major difference in reading the data into Excel, Pandas, numpy, etc....

That said, if the NONE representation shows up in other data sets, a format that uses dummy indices or another approach to facilitate regular columns would be worthwhile I think

**Comment by aleaf** [comment]
without knowing all of the implications, it seems like it would make sense to have this apply to any list-type input that currently uses NONE.

**Comment by langevin-usgs** [comment]
What about a to_df() method in flopy?  Seems like this might be a better place to handle this than in MODFLOW?

**Comment by mnfienen-usgs** [comment]
I disagree because so much of MF6 I/O is easier to comprehend without needing flopy at all. It's enticing to have a tabular dataset and will encourage people to read directly with typical tools only to be surprised by the NONE creeping in there. That's my thought anyway

**Comment by langevin-usgs** [comment]
Plan is to no longer support NONE for CELLID.  This was fixed with #1311.  Release notes for the next version will indicate: "Added functionality to support zero values for each grid dimension when specifying the CELLID for SFR reaches that are not connected to an underlying groundwater grid cell. For example, for a DIS grid a CELLID of 0 0 0 should be specified for unconnected reaches. Warning messages will be issued if NONE is specified for unconnected reaches. Specifying a CELLID of NONE will eventually be deprecated and will cause MODFLOW 6 to terminate with an error."
